README.md

```diff
@@ -105,6 +105,68 @@ fastify.post('/', function (req, reply) {
 })
 ```
 
+You can also use all the parsed HTTP request parametes to the body:
+
+```js
+const options = {
+  addToBody: true,
+  onData: (fieldName, data) => {
+    // Manage the file stream like you need
+    // By default the data will be added in a Buffer
+    // Be careful to accumulate the file in memory!
+  }
+  limit: { /*...*/ } // You can the limit options in any case
+}
+
+fastify.register(require('fastify-multipart'), options)
+
+fastify.post('/', function (req, reply) {
+  console.log(req.body)
+  // This will print out:
+  // {
+  //   myStringField: 'example',
+  //   anotherOne: 'example',
+  //   myFilenameField: {
+  //     data: <Buffer>,
+  //     encoding: '7bit',
+  //     filename: 'README.md',
+  //     limit: false,
+  //     mimetype: 'text/markdown'
+  //   }
+  // }
+
+  reply.code(200).send()
+})
+```
+
+NB: The `req.body.<fieldName>.data` will be an empty array if you configure the `onData` option.
+
+With this setting you will be able to apply the validation to your service like this:
+
+```js
+fastify.post('/upload', {
+  schema: {
+    body: {
+      type: 'object',
+      required: ['myStringField', 'myFilenameField'],
+      properties: {
+        myStringField: { type: 'string' },
+        myFilenameField: {
+          type: 'object',
+          properties: {
+            encoding: { type: 'string' },
+            filename: { type: 'string' },
+            limit: { type: 'boolean' },
+            mimetype: { type: 'string' }
+          }
+        }
```

Comment:
I'm not convinced by the fact that these properties are defined by the plugin and so they will be fixed for the user. Maybe we can use a shared schema that is added by the plugin? 

These will also show up in swagger - and this is really not ideal. I don't now how to handle these.

Reply:
> Maybe we can use a shared schema that is added by the plugin?

This could be helpful! So the user must set only a set of key-value:
```js
{
  myStringField: { type: 'string' },
  myFilenameField: 'fileType#',
  other: { type: 'number' },
  ...
}
```

> These will also show up in swagger - and this is really not ideal. I don't now how to handle these.

In the first place we can hide the file properties: now we are not supporting the `type: file`, so I think that when we will support it we can fix this plugin.
I'm working on the swagger side too but I wouldn't add to this PR also those considerations.
In `fastify-swagger` the work will take much longer

Reply:
Let's do the shared schema then, something like "MultipartFileType" or something that we can use to avoid collision with a user type.

Reply:
I think I have done 👍